### DISCUSSION TITLE
[BUG]:ilogtail 输入类型为kafka不能使用过滤插件过滤数据

### DISCUSSION THREAD

**Discussion by codingdancer** [alibaba/ilogtail#665] [Help] [answered]
**Describe the bug**
A clear and concise description of what the bug is, ideally within 20 words.
ilogtail 输入类型为kafka不能使用过滤插件过滤数据

**iLogtail Running Environment**
Please provide the following information:

- ilogtail version:
  1.1.1
- Yaml configuration:
![image](https://user-images.githubusercontent.com/102849812/219364136-42e13426-6f4d-4031-a1ce-dd5071666d66.png)

- ilogtail.LOG:

- logtail_plugin.LOG:
![image](https://user-images.githubusercontent.com/102849812/219364240-c4a73640-9596-4675-8aa9-0a607098434a.png)

**Comment by linrunqi08** [comment]
可以用1.3.1的版本测试一下吗

**Reply by codingdancer** [reply]
> 可以用1.3.1的版本测试一下吗

看过这一块的源码了，是一样的，消息转换的时候key是kafka消息的id，所以每条消息的key都不一样

![image](https://user-images.githubusercontent.com/102849812/220572963-edf7f9cc-637a-4704-89ea-9def43777d23.png)

**Comment by linrunqi08** [comment, marked as answer]
@codingdancer  这里使用message.key作为key确实不太合理，可以提交一个PR吗？考虑到兼容性，这里建议在配置中增加一个配置项，默认还是维持原来的行为，打开配置项之后，msg.Key写入到key字段，msg.Value写入到value字段
